ably/realtime_channel_spec_test.go

```diff
@@ -155,6 +157,1258 @@ func attachAndWait(t *testing.T, channel *ably.RealtimeChannel) {
+		}
+
+		defer safeclose(t, close)
+
+		ablytest.Soon.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+	})
+
+	t.Run("RTL4j1: AttachResume should be True when Attached", func(t *testing.T) {
+		t.Parallel()
+
+		app, err := ablytest.NewSandbox(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer safeclose(t, app)
+
+		c, close := TransitionConn(t, nil, app.Options()...)
+		defer safeclose(t, close)
+
+		close = c.To(
+			connecting,
+			connected,
+		)
+
+		defer safeclose(t, close)
+
+		channelTransitioner := c.Channel("test")
+		channel := channelTransitioner.Channel
+
+		channelStateChanges := make(ably.ChannelStateChanges, 10)
+		channel.OnAll(channelStateChanges.Receive)
+
+		// channel state is initialized
+		if expected, got := ably.ChannelStateInitialized, channel.State(); expected != got {
+			t.Fatalf("expected %v; got %v", expected, got)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is INITIALIZED")
+		}
+
+		channelTransitioner.ToSpecifiedState(chAttaching, chAttached)
+
+		var channelStatechange ably.ChannelStateChange
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+		if channel.GetAttachResume() != true {
+			t.Fatalf("Channel attach resume should be true when channel state is ATTCHED")
+		}
+
+		channelTransitioner.ToSpecifiedState(chDetaching)
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateDetaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is DETACHING")
+		}
+
+		channelTransitioner.ToSpecifiedState(chFailed)
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateFailed, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is FAILED")
+		}
+	})
+
+	t.Skip("RTL4j2: Rewind flag should allow to receive historic messages")
```

Comment:
This should be removed.

Reply:
It's skipped, as per spec we need to implement this test, so we can't remove it for the time being